_posts/2020-06-17-#18242.md

```diff
@@ -10,14 +10,80 @@ status: upcoming
+  peers. A malicious infrastructure entity could use this to influence the
+  network topology, determine the origin of transactions, or perform attacks
+  against offchain protocols.
+
+- BIP 324 is the second proposal for an encryption standard for the Bitcoin P2P
+  network. The previous proposal ([BIP
+  151](https://github.com/bitcoin/bips/blob/master/bip-0151.mediawiki)) was
+  also authored by Jonas Schnelli and has been withdrawn.
+
+- In BIP 324, messages are encrypted by the stream cipher
+  [ChaCha20](https://tools.ietf.org/html/rfc7539) with a Poly1305 [Message
+  Authentication
+  Code](https://en.wikipedia.org/wiki/Message_authentication_code).
+
+- A BIP 324 session starts with an [Eliptic Curve Diffie Hellman key
+  exchange](https://en.wikipedia.org/wiki/Elliptic-curve_Diffie%E2%80%93Hellman)
+  to establish a shared session key between peers (this key exchange is not yet
+  implemented). From this shared sesssion secret, 2 keys, `K_1` and `K_2` are
+  derived. `K_1` is used to encrypt the 3 bytes packet length. `K_2` is used to
+  encrypt and authenticate the rest of the packet. Using symmetric keys `K_1` and
+  `K_2`, the receiver first decrypts the length number and from this offset
+  authenticates the packet. If authentication succeeds, the receiver decrypts the
+  message payload and hands the content to the processing layer.
+
+- Other PRs that are part of BIP 324 implementation:
+  - [#15206: Immediately disconnect on invalid net message checksum](https://github.com/bitcoin/bitcoin/pull/15206) (discussed in a [previous Bitcoin Core PR Review Club](/15206))
+  - [#15519: Add Poly1305 implementation](https://github.com/bitcoin/bitcoin/pull/15519)
+  - [#15649: Add ChaCha20Poly1305@Bitcoin AEAD](https://github.com/bitcoin/bitcoin/pull/15649)
+  - [#16202: Refactor network message deserialization](https://github.com/bitcoin/bitcoin/pull/16202)
+  - [#16562: Refactor message transport packaging](https://github.com/bitcoin/bitcoin/pull/16562)
+
+- [Jonas Schnelli's talk](https://github.com/kanzure/diyhpluswiki/blob/master/transcripts/breaking-bitcoin/2019/p2p-encryption.mdwn)
+  at Breaking Bitcoin 2019 has additional information on BIP 324.
+
+- A further step to prevent eavesprodding on the P2P network would be to add
+  peer authentication, which is outside the scope of BIP 324. How to
+  authenticate peers in an anonymous network is an [area of active
+  research](https://gist.github.com/sipa/d7dcaae0419f10e5be0270fada84c20b).
+
 ## Questions
 
---->
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK?](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)
+   (Don't forget to put your PR review on GitHub.)
+
+2. What's your process to review a PR implementing a new BIP? Did you read the
+   BIP or the code first? How can you ensure that BIP is correct? How can you ensure
+   that code implementats the BIP a correctly?
```

Comment:
*implements

Reply:
thanks. Fixed this and a few other typos